cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/MonitorHandlerTest.java

```diff
@@ -97,4 +99,27 @@ public void testInstances() throws Exception {
     Assert.assertEquals(1, (int) result.get("requested"));
     Assert.assertEquals(1, (int) result.get("provisioned"));
   }
+
+  @Test
+  public void testRestartInstances() throws Exception {
+    String path = String.format("system/services/%s/instances/restart", Constants.Service.APP_FABRIC_HTTP);
+    HttpURLConnection urlConn = openURL(path, HttpMethod.PUT);
+
+    Assert.assertEquals(HttpResponseStatus.OK.getCode(), urlConn.getResponseCode());
+
+    TimeUnit.SECONDS.sleep(5L);
+
+    path = String.format("system/services/%s/instances/restart", Constants.Service.APP_FABRIC_HTTP);
+    urlConn = openURL(path, HttpMethod.GET);
+
+    Assert.assertEquals(HttpResponseStatus.OK.getCode(), urlConn.getResponseCode());
+
+    RestartServiceInstancesStatus result =
+      GSON.fromJson(new String(ByteStreams.toByteArray(urlConn.getInputStream()), Charsets.UTF_8),
+                    RestartServiceInstancesStatus.class);
```

Comment:
The other method is the test is using GSON so lets keep it that way and address it in separate PR.